### PR TITLE
SONARPY-2355 Update licenses to SSALv1

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Context for the format update of `license-maven-plugin`: see this [PR](https://github.com/SonarSource/parent-oss/pull/215). Not respecting the new format prevented our custom configuration from being taken into account (Python files were neither checked nor updated).

Currently, SSALv1 itself is flagged as a forbidden dependency for our internal checks. Because of this, they are overridden in our checks (see [BUILD-7017](https://sonarsource.atlassian.net/browse/BUILD-7017)).

[BUILD-7017]: https://sonarsource.atlassian.net/browse/BUILD-7017?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,4 +82,6 @@ To fix that please execute: `mvn license:format`.
 
 Copyright 2011-2024 SonarSource.
 
-Licensed under the [GNU Lesser General Public License, Version 3.0](http://www.gnu.org/licenses/lgpl.txt)
+SonarQube analyzers released after November 29, 2024, including patch fixes for prior versions, are published under the Sonar Source-Available License Version 1 (SSALv1).
```
Comment (Seppli11) [inline README.md:85]: According to SQRP-80, the readme should link to the LICENSE.txt file

**Review by Seppli11 — APPROVED** [review]
Apart from the missing link to the license file, it looks good to me.

SQRP-80 mentions that projects that should remain under LGPL need to modify the pom.xml accordingly. But this doesn't apply to any of the projects in this repo, right?

**Comment by guillaume-dequenne-sonarsource** [comment]
> Apart from the missing link to the license file, it looks good to me.
> 
> SQRP-80 mentions that projects that should remain under LGPL need to modify the pom.xml accordingly. But this doesn't apply to any of the projects in this repo, right?

It would apply to the `sonar-scanner-python`, if it were a maven project, but it's not so it should be okay.

That said, the serializer under this repo should also move to SSALv1, and I realize my initial changes didn't account for that.

**Review by Seppli11 — APPROVED** [review]
LGTM

**Pull request merged**